src/assets/scripts/client/aircraft/FlightManagementSystem/WaypointModel.js

```diff
@@ -0,0 +1,44 @@
+import _get from 'lodash/get';
+
+/**
+ *
+ * @class WaypointModel
+ */
+export default class WaypointModel {
+    /**
+     *
+     * @constructor
+     * @param waypointProps {object}
+     */
+    constructor(waypointProps) {
+        this.name = '';
+        this.position = null;
+        this.speedRestriction = -1;
+        this.altitudeRestriction = -1;
+
+        // this.hold = {
+        //     dirTurns: null,
+        //     fixName: null,
+        //     fixPos: null,
+        //     inboundHd: null,
+        //     legLength: null,
+        //     timer: 0
+        // };
```

Comment:
ha!  well, not yet.  I'm not sure this is really a poor implementation, I mean it works doesn't it?  Once I get the main methods for the Fms and a basic Mcp up and running, holds are going to be the first thing I tackle.  So we'll know soon.